transport_test.go

```diff
@@ -125,6 +126,87 @@ func TestGetRequestBodyFromEventCompletelyInvalid(t *testing.T) {
 	}
 }
 
+func TestGetEnvelopeFromBody(t *testing.T) {
+	body := getRequestBodyFromEvent(&Event{
+		Message: "testing",
+	})
+	env := getEnvelopeFromBody(body)
+	envString := env.String()
+	envParts := strings.Split(envString, "\n")
+
+	if len(envParts) != 4 {
+		t.Errorf("\"%s\" should have len of 4, instead has len of %d", envParts, len(envParts))
+	}
+
+	if envParts[3] != "" {
+		t.Errorf("\"%s\" is not empty", envParts[3])
+	}
+
+	var header map[string]interface{}
+	err := json.Unmarshal([]byte(envParts[0]), &header)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sentAtStr := header["sent_at"].(string)
+	if sentAtStr == "" {
+		t.Error("sent_at was not set in header")
+	}
+
+	payload := envParts[2]
+	if payload == "" {
+		t.Error("Item payload is empty")
+	}
```

Comment:
The intention here is to test that the serialized envelope is correct, right?
In that case, we don't need / should not be trying to do a roundtrip and deserialize the bytes.

Simply compare the bytes to expected output.

Reply:
The problem here is that `sentry_at` is generated using `time.Now()`, so I can't just compare strings. Can def compare bytes for the other parts of the envelope though.

Reply:
I changed this so that we split it up to just check if `sent_at` is set, and then we use `cmp.Diff` to compare the rest of the payload.

Reply:
This sounds like the typical case to fake time in tests :)

Reply:
Since there are many ways to go about it, I'll give an explicit suggestion on site (the function to create the request body / envelope).